src/ServiceAccountSignerTrait.php

```diff
@@ -37,11 +38,9 @@ public function signBlob($stringToSign, $forceOpenssl = false)
         $privateKey = $this->auth->getSigningKey();
 
         $signedString = '';
-        if (class_exists('\\phpseclib\\Crypt\\RSA') && !$forceOpenssl) {
-            $rsa = new RSA();
-            $rsa->loadKey($privateKey);
-            $rsa->setSignatureMode(RSA::SIGNATURE_PKCS1);
-            $rsa->setHash('sha256');
+        if (class_exists(phpseclib3\Crypt\RSA::class) && !$forceOpenssl) {
+            $key = PublicKeyLoader::load($privateKey);
+            $rsa = $key->withHash('sha256')->withPadding(RSA::SIGNATURE_PKCS1);
```

Comment:
The methods are sourced from here: [PrivateKey.php#L447-L452](https://github.com/phpseclib/phpseclib/blob/8516c99f9e4261c634a59c2c28ddfaa8d15726fb/phpseclib/Crypt/RSA/PrivateKey.php#L447-L452) and [PrivateKey.php#L281](https://github.com/phpseclib/phpseclib/blob/8516c99f9e4261c634a59c2c28ddfaa8d15726fb/phpseclib/Crypt/RSA/PrivateKey.php#L281)